src/textUtils.c

```diff
@@ -105,3 +105,48 @@ size_t str_formatMetadata(const uint8_t* metadataHash, size_t metadataHashSize,
 {
 	return encode_hex(metadataHash, metadataHashSize, out, outSize);
 }
+
+// TODO improve this
+size_t urlToBuffer(const char* url, uint8_t* buffer, size_t bufferSize)
+{
+	size_t urlLength = strlen(url);
+	ASSERT(urlLength < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize >= urlLength);
+
+	const char* validChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~:/?#[]@!$&'()*+,;%=";
+	size_t validCharsLength = strlen(validChars);
+	for (size_t i = 0; i < urlLength; i++) {
+		bool valid = false;
+		for (size_t j = 0; j < validCharsLength; j++) {
+			if (url[i] == validChars[j]) {
+				valid = true;
+				break;
+			}
+		}
+		if (!valid)
+			THROW(ERR_INVALID_DATA);
+
+		buffer[i] = url[i];
+	}
+
+	return urlLength;
+}
+
+// TODO improve this?
+size_t dnsNameToBuffer(const char* dnsName, uint8_t* buffer, size_t bufferSize)
+{
+	size_t dnsNameLength = strlen(dnsName);
+	ASSERT(dnsNameLength < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize >= dnsNameLength);
+
+	for (size_t i = 0; i < dnsNameLength; i++) {
+		if (dnsName[i] > 127)
```

Comment:
aren't there the same restrictions on character set as for urls?

Reply:
I don't know --- albeit they call it `dns_name` in CDDL, it is not a name, but some type of record, e.g. a space is allowed, probably some forms of newlines, too.

I don't feel it is worth spending too much time on this --- we can't do full validation because it is too complex (and rather annoying to do in C, especially with limited memory), and even a small mistake might lead to denying valid transactions. We don't even display the relays. So I would let the full node decide what is valid and what is not.

We can discuss it at a standup.

Reply:
allow only reasonable part of ascii - something between 32-126 (http://www.asciitable.com/). Basically, just validate that we can at least display it